pkg/cmd/experimental/diagnostics/host.go

```diff
@@ -0,0 +1,76 @@
+package diagnostics
+
+import (
+	"fmt"
+	"github.com/openshift/origin/pkg/diagnostics/host"
+	systemddiagnostics "github.com/openshift/origin/pkg/diagnostics/systemd"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+	"k8s.io/kubernetes/pkg/util"
+	"os"
+)
+
+const (
+	StandardMasterConfigPath string = "/etc/openshift/master/master-config.yaml"
+	StandardNodeConfigPath   string = "/etc/openshift/node/node-config.yaml"
+)
+
+var (
+	AvailableHostDiagnostics = util.NewStringSet("AnalyzeLogs", "UnitStatus", "MasterConfigCheck", "NodeConfigCheck")
+)
+
+func (o DiagnosticsOptions) buildHostDiagnostics() ([]types.Diagnostic, bool /* ok */, error) {
+	requestedDiagnostics := intersection(util.NewStringSet(o.RequestedDiagnostics...), AvailableHostDiagnostics).List()
+	if len(requestedDiagnostics) == 0 { // no diagnostics to run here
+		return nil, true, nil // don't waste time on discovery
+	}
+	isHost := o.IsHost
+	// check for standard host config paths if not given
+	if len(o.MasterConfigLocation) == 0 {
+		if _, err := os.Stat(StandardMasterConfigPath); !os.IsNotExist(err) {
+			o.MasterConfigLocation = StandardMasterConfigPath
+			isHost = true
+		}
+	} else {
+		isHost = true
+	}
+	if len(o.NodeConfigLocation) == 0 {
+		if _, err := os.Stat(StandardNodeConfigPath); !os.IsNotExist(err) {
+			o.NodeConfigLocation = StandardNodeConfigPath
+			isHost = true
+		}
+	} else {
+		isHost = true
+	}
+
+	// If we're not looking at a host, don't try the diagnostics
+	if !isHost {
+		return nil, true, nil
+	}
+
+	diagnostics := []types.Diagnostic{}
+	systemdUnits := systemddiagnostics.GetSystemdUnits(o.Logger)
+	for _, diagnosticName := range requestedDiagnostics {
+		switch diagnosticName {
+		case "AnalyzeLogs":
```

Comment:
Yes, I'm going through and changing all the diagnostic name strings to constants.